examples/java/src/test/java/org/apache/beam/examples/cookbook/CombinePerKeyExamplesIT.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.examples.cookbook;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.api.client.util.BackOff;
+import com.google.api.client.util.BackOffUtils;
+import com.google.api.client.util.Sleeper;
+import com.google.api.services.bigquery.model.QueryResponse;
+import org.apache.beam.sdk.extensions.gcp.options.GcpOptions;
+import org.apache.beam.sdk.extensions.gcp.util.BackOffAdapter;
+import org.apache.beam.sdk.io.gcp.testing.BigqueryClient;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.testing.TestPipelineOptions;
+import org.apache.beam.sdk.util.FluentBackoff;
+import org.joda.time.Duration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** An end-to-end test for {@link org.apache.beam.examples.cookbook.CombinePerKeyExamples}. */
+@RunWith(JUnit4.class)
+public class CombinePerKeyExamplesIT {
+  private CombinePerKeyExamplesOptions options;
+  private final String timestamp = Long.toString(System.currentTimeMillis());
+  private final String outputDatasetId = "combine_per_key_examples" + timestamp;
+  private final String outputTable = "combine_per_key_examples_table";
+  private final Long defaultExpiration = 1000L * 60 * 60;
+  private String projectId;
+  private BigqueryClient bqClient;
+
+  /** Options for the CombinePerKeyExamples Integration Test. */
+  public interface CombinePerKeyExamplesOptions
+      extends TestPipelineOptions, CombinePerKeyExamples.Options {}
+
+  @Before
+  public void setupTestEnvironment() throws Exception {
+    PipelineOptionsFactory.register(TestPipelineOptions.class);
+    this.options = TestPipeline.testingPipelineOptions().as(CombinePerKeyExamplesOptions.class);
+    this.projectId = TestPipeline.testingPipelineOptions().as(GcpOptions.class).getProject();
+    this.bqClient = new BigqueryClient("CombinePerKeyExamplesIT");
+    this.bqClient.createNewDataset(this.projectId, this.outputDatasetId, defaultExpiration);
```

Comment:
Is there a benefit to creating/deleting the datasets? It looks like some integration tests reuse the same dataset with a table expiration

Reply:
I have seen both scenarios, for example, this existing test 

https://github.com/apache/beam/blob/master/examples/java/src/test/java/org/apache/beam/examples/complete/TrafficMaxLaneFlowIT.java#L63

I think creating and deleting a dataset inside the test could help to avoid side effects because of leftover data, but I don't have a strong preference.